persistence-modules/spring-data-jpa-query-4/src/main/java/com/baeldung/spring/data/jpa/maxvalue/EmployeeMaxValueService.java

```diff
@@ -0,0 +1,30 @@
+package com.baeldung.spring.data.jpa.maxvalue;
+
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.Optional;
+
+@Service
+public class EmployeeMaxValueService {
+    private final EntityManager entityManager;
+
+    public EmployeeMaxValueService(EntityManager entityManager) {
```

Comment:
Should we have `@Autowired` on the constructor for clarity about where the dependency comes from?

```suggestion
    @Autowired
    public EmployeeMaxValueService(EntityManager entityManager) {
```